### PR TITLE
libexosip: 4.1.0 -> 5.2.0

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Motivation for this change

Fix CVE-2014-10375.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/exosip/default.nix [hunk 1]
```diff
@@ -2,23 +2,16 @@
...
-    sha256 = "17cna8kpc8nk1si419vgr6r42k2lda0rdk50vlxrw8rzg0xp2xrw";
+    url = "mirror://savannah/exosip/${pname}-${version}.tar.gz";
+    sha256 = "09bj7cm6mk8yr68y5a09a625x10ql6an3zi4pj6y1jbkhpgqibp3";
   };
 
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [ libosip openssl ];
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/exosip/default.nix:16]: Please remove fetchpatch from inputs.

File: pkgs/development/libraries/exosip/default.nix [hunk 2]
```diff
@@ -2,23 +2,16 @@
...
-      url = "https://sources.debian.net/data/main/libe/libexosip2/4.1.0-2.1/debian/patches/openssl110.patch";
-      sha256 = "01q2dax7pwh197mn18r22y38mrsky85mvs9vbkn9fpcilrdayal6";
-    })
-  ];
-
   meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/exosip/default.nix:15]:
```suggestion
  meta = with lib; {
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110231` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>sipwitch: log <a href="https://termbin.com/vtab">https://termbin.com/vtab</a></li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>libexosip</li>
<li>liblinphone</li>
<li>linphone</li>
</ul>
</details>

**Comment by LeSuisse** [comment]
> Either update sipwitch, too or try to find a patch.

Sorry I missed that, I do not have `NIXPKGS_ALLOW_INSECURE=1` set so it seems that my local run of nixpkgs-review did not show me the issue.

I'm not sure what to do here. The breaking change in the API of libexosip (`eXosip_call_build_prack`) does not seem trivial to fix, the upstream project seems dead (no commits since 2015 and barely any activity on the mailing list since 2016) and the maintainer of the package has step out in 2018 (66d7126255b96a3d9c1cc0cdd7c2cd79d1789758). Since liblinphone/linphone  which are depending upon libexosip are still well alive maybe it is worth to drop the sipswitch package so we can move forward.
What do you think?

**Comment by SuperSandro2000** [comment]
Oh, sorry I had allow insecure still on from texting earlier. I think it would be safe to mark the insecure package broken and then merge this PR.

**Comment by SuperSandro2000** [comment]
@LeSuisse please fix the merge conflict.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110231` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>sipwitch</li>
</ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>libexosip</li>
<li>liblinphone</li>
<li>linphone</li>
</ul>
</details>

**Pull request merged**